integration-tests/opentelemetry-reactive/src/test/java/io/quarkus/it/opentelemetry/reactive/OpenTelemetryWithSpanAtStartupTest.java

```diff
@@ -38,6 +40,14 @@ public class OpenTelemetryWithSpanAtStartupTest {
 
     @Test
     void testGeneratedSpansUsingRestClientReactive() {
+
+        // There's a bit of asynchronousness in recording and exporting the spans, so to avoid issues, give the startup spans a (small) moment to arrive
+        await().atMost(Duration.ofSeconds(5L)).pollDelay(Duration.ofMillis(50)).until(() -> {
+            // make sure spans are cleared
+            List<Map<String, Object>> spans = getSpans();
```

Comment:
We are waiting for 2 spans, we should hold until `spans.size` == 2

Reply:
Probably better to wait until `size >=2`, right? Otherwise if 3 spans come in, the test will get stuck in the await loop, when it should be going on and failing in the assertion. 

I guess the other question is, if one span comes in, and there's a really long delay for the next span, is that a problem? With my version it would fail in that case, and with `size >=2`, it would pass. I don't know what the best expectation is?

Reply:
I've also just spotted the`// make sure spans are cleared` comment isn't right, it should be something like `// make sure incoming spans are processed`, so I'll adjust.

Reply:
I've set the wait to wait for 2, adjusted the comment, squashed and pushed.

Reply:
Totally agreed that the test should fail if it gets 3 spans. The question is just "how does it fail? Does it wait for 10s hoping three spans turns into 2 spans and fail with an awaitility error, or do we fail on the assertion that says "there should be 2 spans?"

I think it's more explicit and more helpful to fail on the existing assertion; the await() block at the beginning isn't really intended to describe expected behaviour, it's just trying to pause to make sure there's no cross-talk between tests.

Reply:
I'm fairly sure that what I'm seeing otherwise *is* cross-talk, because removing the `reset()` block from the other test changes the behaviour of this test.

Reply:
>  "there should be 2 spans?"

Yes, and ideally we should print all the spans that were found to understand what happened.

Reply:
I think 10s wait is too much. 5s is usually what I use. 
It's more than enough to receive in memory spans.

Reply:
> > "there should be 2 spans?"
> 
> Yes, and ideally we should print all the spans that were found to understand what happened.

Right, that's why I didn't want the failure to happen in the `await` block if there's three spans. If there's 3 spans, the execution should leave the `await` block and fail here: 

```
        List<Map<String, Object>> spans = getSpans();
        assertEquals(2, spans.size());
```

The await block isn't intended to be making assertions about the behaviour, it's just intended to introduce a pause that's long enough for asynchronous processing to happen.

Reply:
ummm... I wasn't aware of the fail-fast option, maybe it can be used here:
example:
```
await().timeout(Duration.ofSeconds(5))
       .failFast(() -> orders.findById(1234).state == PAID)
       .until(() -> orders.findById(1234).state == VOIDED);
```

Reply:
That would work, but why bother adding the extra complexity? Why not just say "I want to start making assertions about the spans once some spans have arrived?"

The await() isn't replacing the assertions we already have, it's just saying "hey, maybe pause until we have spans before looking at the spans."

Reply:
With the code as written, the test will fail quickly if there are more than 2 spans. So I'm not sure what problem we're trying to fix.

Reply:
I'm convinced. That makes sense.
Thanks @holly-cummins

Reply:
Thanks!